### ISSUE TITLE
Application v 3.5.3 damaged

### ISSUE THREAD

**Issue by MarkDurre** [veusz/veusz#630]
Hi, I'm getting the error "Veusz.app" isdamaged and can't be opened. You should move it to the bin" when I download the dmg from Github.

**Comment by MarkDurre** [comment]
On MacOS Catalina 10.15.7

**Comment by korintje** [comment]
I also heard this issue from other people.

The following command worked well as a work around of the problem.

```
xattr -d com.apple.quarantine Veusz.app
```

It may relate to Gatekeeper of macOS.

**Comment by MarkDurre** [comment]
Manu thanks – that worked!

Dr. Mark Durré
Centre for Astrophysics and Supercomputing
Swinburne University of Technology
PO Box 218, Hawthorn
Victoria 3122, Australia
Email: ***@***.******@***.***>


From: Takuro Hosomi ***@***.***>
Reply to: veusz/veusz ***@***.***>
Date: Friday, 4 November 2022 at 1:43 pm
To: veusz/veusz ***@***.***>
Cc: Mark Durre ***@***.***>, Author ***@***.***>
Subject: Re: [veusz/veusz] Application v 3.5.3 damaged (Issue #630)


I also heard this issue from other people.

The following command worked well as a work around of the problem.

xattr -d com.apple.quarantine Veusz.app<http://Veusz.app>

It may relate to Gatekeeper of macOS.

**Comment by jeremysanders** [comment]
If we wanted to fix this properly, presumably someone would have to pay the $99/yr Apple tax?

If there's a Mac developer who's already paying this and would like to step up and take the task of building on MacOS, that would be great.

**Comment by korintje** [comment]
# Issue summary

## TL;DR
1. Open the terminal
2. Download .dmg file by curl: `curl -OL "https://github.com/veusz/veusz/releases/download/veusz-3.5.3/veusz-3.5.3-AppleOSX.dmg"`
3. Install Veusz with the downloaded .dmg file

## Explanation
The issue comes from "Gatekeeper" of macOS. 
Gatekeeper requires all apps to be signed. Developers must pay $99/year to Apple to sign.
https://www.cnet.com/tech/computing/how-to-bypass-damaged-application-warnings-in-os-x/

Users can manually remove extended attribute `com.apple.quarantine`, which says the file is signed or not.
Command: `xattr -rc "veusz-3.5.3-AppleOSX.dmg"`

Interestingly, the attribute will be given only when files are downloaded by Safari. Therefore, if you download the files by a command like `curl`, you don't have to do anything additionally.



## Appendix
I personally use my own installation [script](https://github.com/korintje/veusz-installer-macOS/blob/main/README.md) on macOS. This script automatically downloads and installs the latest Veusz at the time bypassing the above issue.


>If we wanted to fix this properly, presumably someone would have to pay the $99/yr Apple tax?

Unfortunately, their answer seems to be "yes", unless the app officially from an educational organization.
(https://developer.apple.com/support/membership-fee-waiver/)

**Comment by jeremysanders** [comment]
Could people affected by this please try out the test version 3.5.99.1 here?
  https://github.com/veusz/veusz/releases/tag/veusz-3.5.99

Is it the same or better?

**Comment by korintje** [comment]
> Could people affected by this please try out the test version 3.5.99.1 here? https://github.com/veusz/veusz/releases/tag/veusz-3.5.99
> 
> Is it the same or better?

I have tried ver. 3.5.99.1 but faced the same error.
--------------------------------------
Mac mini (Late 2014)
macOS Monterey

**Comment by jeremysanders** [comment]
Ok, I think I've worked out what was causing the error. I was modifying the info.plist in the bundle after it was built by pyinstaller. I'm now rerunning the self-signing after doing that. @korintje - could you try 3.5.99.3 from the 3.5.99 link above and see whether the situation is now better?

**Comment by korintje** [comment]
@jeremysanders 
I confirmed that Veusz 3.5.99.3 can be launched successfully at MacOS without any tricks.
That worked. Thank you so much!

---
For someone who find this conversation by some searches:
Only the first time after you install the program, you need to select "open" from the context menu that comes up when you press "Option + right click" on the app icon.

**Comment by jeremysanders** [comment]
Thanks for checking. I'll close this now.

**Comment by jeremysanders** [comment]
PS I've made a new release with this fix in it.
